src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/RazorSemanticTokensLegend.cs

```diff
@@ -2,107 +2,35 @@
-
-    public int RazorComment => RazorTokenTypeMap[RazorCommentType];
-    public int RazorCommentStar => RazorTokenTypeMap[RazorCommentStarType];
-    public int RazorCommentTransition => RazorTokenTypeMap[RazorCommentTransitionType];
-    public int RazorComponentAttribute => RazorTokenTypeMap[RazorComponentAttributeType];
-    public int RazorComponentElement => RazorTokenTypeMap[RazorComponentElementType];
-    public int RazorDirective => RazorTokenTypeMap[RazorDirectiveType];
-    public int RazorDirectiveAttribute => RazorTokenTypeMap[RazorDirectiveAttributeType];
-    public int RazorDirectiveColon => RazorTokenTypeMap[RazorDirectiveColonType];
-    public int RazorTagHelperAttribute => RazorTokenTypeMap[RazorTagHelperAttributeType];
-    public int RazorTagHelperElement => RazorTokenTypeMap[RazorTagHelperElementType];
-    public int RazorTransition => RazorTokenTypeMap[RazorTransitionType];
-
-    public SemanticTokensLegend Legend => _lazyInitializer.Value.Legend;
-
-    private Dictionary<string, int> RazorTokenTypeMap => _lazyInitializer.Value.TokenTypeMap;
-
-    private readonly Lazy<(SemanticTokensLegend Legend, Dictionary<string, int> TokenTypeMap)> _lazyInitializer;
-
-    [ImportingConstructor]
-    public RazorSemanticTokensLegendService(IClientCapabilitiesService clientCapabilitiesService)
+    public Types TokenTypes => _typesLazy.Value;
+    public Modifiers TokenModifiers { get; } = new Modifiers();
+    public SemanticTokensLegend Legend => new SemanticTokensLegend()
     {
-        // DI calls this constructor to build the service container, but we can't access clientCapabilitiesService
-        // until the language server has received the Initialize message, so we have to do this lazily.
-        _lazyInitializer = new Lazy<(SemanticTokensLegend, Dictionary<string, int>)>(() =>
-        {
-            using var _ = ArrayBuilderPool<string>.GetPooledObject(out var builder);
-
-            var supportsVsExtensions = clientCapabilitiesService.ClientCapabilities.SupportsVisualStudioExtensions;
-
-            builder.AddRange(RazorSemanticTokensAccessor.GetTokenTypes(supportsVsExtensions));
-
-            var razorTokenTypeMap = new Dictionary<string, int>();
-            foreach (var razorTokenType in GetRazorSemanticTokenTypes())
-            {
-                razorTokenTypeMap.Add(razorTokenType, builder.Count);
-                builder.Add(razorTokenType);
-            }
-
-            var legend = new SemanticTokensLegend()
-            {
-                TokenModifiers = s_tokenModifiers,
-                TokenTypes = builder.ToArray()
-            };
+        TokenModifiers = TokenModifiers.TokenModifiers,
+        TokenTypes = _typesLazy.Value.TokenTypes
+    };
 
-            return (legend, razorTokenTypeMap);
-        });
-    }
-
-    private static ImmutableArray<string> GetRazorSemanticTokenTypes()
+    private static ImmutableArray<string> GetStaticPropertyValues(Type type)
```

Comment:
>GetStaticPropertyValues [](http://example.com/codeflow?start=42&length=23)

nit: if we do keep this (see my other comment), should this be Fields instead of Properties?

Reply:
Whoops, yes, good catch